app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java

```diff
@@ -107,40 +115,65 @@ public void onReceive(Context context, Intent intent) {
-
         return rootView;
     }
 
-    public void putMyLocationPin(LocationSample location) {
-        PointSymbol pointSymbol = new PointImageSymbol(
-                getString(R.string.viewer_self_icon_assets_path), 36, 36);
-        putLocationPin(getString(R.string.viewer_my_location_name), location.getLocation(),
-                pointSymbol);
+    @OnClick(R.id.message_fab)
+    public void openSendMessageDialog() {
+        new SendMessageDialogFragment().show(getFragmentManager(), "sendMessageDialog");
     }
 
-    public void putUserLocationPin(String id, PointGeometry pg) {
-        PointSymbol pointSymbol = new PointTextSymbol(EntitiesHelperUtils.getRandomCssColorStirng(),
-                id, 48);
-        putLocationPin(id, pg, pointSymbol);
+    @OnClick(R.id.camera_fab)
+    public void startCameraActivity() {
+        Intent takePictureIntent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+
+        // place where to store camera taken picture
+        try {
+            mImageUri = ImageUtil.getTempImageUri(mApp);
+        } catch (IOException e) {
+            Log.w(TAG_FRAGMENT, "Can't create file to take picture!");
+            return;
+        }
+
+        takePictureIntent.putExtra(MediaStore.EXTRA_OUTPUT, mImageUri);
+        //start camera intent
+        if (takePictureIntent.resolveActivity(mApp.getPackageManager()) != null) {
+            startActivityForResult(takePictureIntent, REQUEST_IMAGE_CAPTURE);
+        }
     }
 
-    private void putLocationPin(String id, PointGeometry pg, PointSymbol pointSymbol) {
-        Entity point = mUsersLocationsLayer.getEntity(id);
-        if (point == null) {
-            point = new Point.Builder()
-                    .setId(id).setGeometry(pg)
-                    .setSymbol(pointSymbol)
-                    .build();
-            mUsersLocationsLayer.addEntity(point);
+    @OnClick(R.id.locate_me_fab)
+    public void zoomToLastLocation() {
+        LocationSample lastKnownLocation = LocationFetcher.getInstance(
+                getActivity()).getLastKnownLocation();
+
+        if (lastKnownLocation == null) {
+            Toast.makeText(getActivity(), R.string.locate_me_fab_no_known_location,
+                    Toast.LENGTH_SHORT).show();
         } else {
-            point.updateGeometry(pg);
+            PointGeometry location = lastKnownLocation.getLocation();
+
+            location.altitude = getResources().getInteger(R.integer.locate_me_button_altitude);
+            mGGMapView.zoomTo(location);
         }
-        if (mGGMapView.getLayer(mUsersLocationsLayer.getId()) == null) {
-            mGGMapView.addLayer(mUsersLocationsLayer);
+    }
+
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if (requestCode == REQUEST_IMAGE_CAPTURE && resultCode == Activity.RESULT_OK) {
+            LocationSample imageLocation = LocationFetcher.getInstance(
+                    getActivity()).getLastKnownLocation();
```

Comment:
I know we all use getActivity().
should we create field mAppContext or something like that and use that instead?